channeldb/channel.go

```diff
@@ -1589,6 +1605,83 @@ func (c *OpenChannel) isBorked(chanBucket kvdb.RBucket) (bool, error) {
 	return channel.chanStatus != ChanStatusDefault, nil
 }
 
+// MarkShutdownSent updates the channel status to indicate that a shutdown has
+// been initiated. It also persists the delivery script that we will use in
+// the Shutdown message so that we can guarantee that the same delivery script
+// is used if a re-connect happens.
+func (c *OpenChannel) MarkShutdownSent(localDeliveryScript []byte,
+	locallyInitiated bool) error {
+
+	return c.markShutdownSent(localDeliveryScript, locallyInitiated)
+}
+
+// markShutdownSent is a helper function which modifies the channel status of
+// the receiving channel and inserts a delivery script under the delivery script
+// key. It adds a status which indicates the party that initiated the channel
+// close.
+func (c *OpenChannel) markShutdownSent(deliveryScript []byte,
+	locallyInitiated bool) error {
+
+	c.Lock()
+	defer c.Unlock()
+
+	putDeliveryScript := func(chanBucket kvdb.RwBucket) error {
+		return chanBucket.Put(deliveryScriptKey, deliveryScript)
+	}
+
+	status := ChanStatusShutdownSent
+	if locallyInitiated {
+		status |= ChanStatusLocalCloseInitiator
+	} else {
+		status |= ChanStatusRemoteCloseInitiator
+	}
+
+	return c.putChanStatus(status, putDeliveryScript)
+}
+
+// DeliveryScript returns the delivery script to use in the channel's Shutdown
+// message. ErrNoDeliveryScript is returned if no such delivery script has been
+// persisted yet.
+func (c *OpenChannel) DeliveryScript() ([]byte, error) {
```

Comment:
blocking: Use something richer than a byte slice as the return type. An Option return would be nice as well and since we already depend on lnwire here and there is a `DeliveryAddress` type there, we can just use that.